remove command fetcher

jrhizor opened the pull request:
resolves https://github.com/airbytehq/airbyte/issues/5428

The previous flow was:
* create a command fetcher pod
* read the output from that pod 
* use the output to create a new pod that actually runs

The new flow is:
* create the pod and have the main container determine if the env variable is set properly or not and start!

This saves a few seconds on every UI operation on Kubernetes! Great idea from @m-ronchi! 

The remaining changes in this PR allow `KubePodProcessIntegrationTest` to run locally on docker-desktop by using `host.docker.internal` as the local ip/host in the tests.

jrhizor commented:
Ready for an initial review. Still need to make sure that we properly escape any entrypoint override string we're passing in.

davinchia commented on airbyte-workers/src/main/java/io/airbyte/workers/process/KubePodProcess.java:
```diff
@@ -120,6 +118,7 @@
   private static final int STDIN_REMOTE_PORT = 9001;
   private static final Map<String, String> AIRBYTE_POD_LABELS = Map.of("airbyte", "worker-pod");
 
+  private final String processRunnerHost;
```
looks like we don't need to store this as class state and having it within the function is sufficient.

davinchia commented on airbyte-workers/src/main/java/io/airbyte/workers/process/KubeProcessFactory.java:
```diff
@@ -47,23 +49,37 @@
+   * @param processRunnerHost is the local host or ip of the machine running the process factory.
+   *        injectable for testing.
    */
   public KubeProcessFactory(String namespace,
```
nit: VisibleForTesting annotation?

davinchia commented on airbyte-workers/src/main/java/io/airbyte/workers/process/KubePodProcess.java:
```diff
@@ -345,7 +305,7 @@ public KubePodProcess(ApiClient officialClient,
         image,
         usesStdin,
-        entrypoint,
+        entrypointOverride,
```
now that we no longer need to specify entrypoint - what is the override for? looks like mostly tests?

jrhizor replied:
This is used by custom operators.

davinchia commented on airbyte-workers/src/main/java/io/airbyte/workers/process/KubePodProcess.java:
```diff
@@ -120,6 +118,7 @@
   private static final int STDIN_REMOTE_PORT = 9001;
   private static final Map<String, String> AIRBYTE_POD_LABELS = Map.of("airbyte", "worker-pod");
```
we forgot to do this when we added the command fetcher pod, can we add a note in the class docstring about the assumption that images need an AIRBYTE_ENTRYPOINT env var set?

davinchia reviewed (approved):
Looks good!

jrhizor commented:
@davinchia PTAL. I addressed comments and added escaping. 

@cgardens and @davinchia I would appreciate a close look at escaping to see if this approach is reasonable.

cgardens reviewed (approved):
🎉 ❤️  awesome. Thank @m-ronchi for calling this out.

cgardens commented:
and yes, the approach to escaping makes sense to me.

michel-tricot commented:
Super nice!

davinchia reviewed (approved):
Nice!
